AlwaysOn/Wearable/src/androidTest/java/com/example/android/wearable/wear/alwayson/MainActivityTests.kt

```diff
@@ -0,0 +1,222 @@
+        onView(withId(R.id.time_stamp)).check(matches(withText(context.getString(R.string.timestamp_label, YEAR_2200))))
+        onView(withId(R.id.state)).check(matches(withText(context.getString(R.string.mode_active_label))))
+        onView(withId(R.id.draw_count)).check(matches(withText(context.getString(R.string.draw_count_label, 1))))
+    }
+
+    @Test
+    fun textIsCorrectAfterFiveSeconds(): Unit = runBlocking {
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        // Advance 5 seconds, one at a time
+        repeat(5) {
+            advanceTime(Duration.ofSeconds(1))
+        }
+
+        onView(withId(R.id.time)).check(matches(withText(FIVE_SEC_DISPLAY)))
+        onView(withId(R.id.time_stamp)).check(matches(withText(context.getString(R.string.timestamp_label, YEAR_2200 + 5000L))))
+        onView(withId(R.id.state)).check(matches(withText(context.getString(R.string.mode_active_label))))
+        onView(withId(R.id.draw_count)).check(matches(withText(context.getString(R.string.draw_count_label, 6))))
+    }
+
+    @Test
+    fun textIsCorrectAfterGoingIntoAmbientMode(): Unit = runBlocking {
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        // Advance 5 seconds, one at a time
+        repeat(5) {
+            advanceTime(Duration.ofSeconds(1))
+        }
+
+        uiDevice.pressKeyCode(KeyEvent.KEYCODE_SLEEP)
+        Thread.sleep(1000) // Ugly sleep, without it sometimes ambient mode won't be entered
+        Espresso.onIdle()
+
+        onView(withId(R.id.time)).check(matches(withText(FIVE_SEC_DISPLAY)))
+        onView(withId(R.id.time_stamp)).check(matches(withText(context.getString(R.string.timestamp_label, YEAR_2200 + 5000L))))
+        onView(withId(R.id.state)).check(matches(withText(context.getString(R.string.mode_ambient_label))))
+        onView(withId(R.id.draw_count)).check(matches(withText(context.getString(R.string.draw_count_label, 7))))
+    }
+
+    @Test
+    fun textIsCorrectAfterGoingIntoAmbientModeAndReceivingIntent(): Unit = runBlocking {
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        // Advance 5 seconds, one at a time
+        repeat(5) {
+            advanceTime(Duration.ofSeconds(1))
+        }
+
+        advanceTime(Duration.ofMillis(500))
+
+        uiDevice.pressKeyCode(KeyEvent.KEYCODE_SLEEP)
+        Thread.sleep(1000) // Ugly sleep, without it sometimes ambient mode won't be entered
+        Espresso.onIdle()
+
+        // Simulate a sent broadcast
+        advanceTime(Duration.ofSeconds(5))
+        scenario.onActivity {
+            PendingIntent.getBroadcast(
+                it, 0, Intent(MainActivity.AMBIENT_UPDATE_ACTION), PendingIntent.FLAG_UPDATE_CURRENT
+            ).send()
+        }
+
+        Espresso.onIdle()
+
+        onView(withId(R.id.time)).check(matches(withText(TEN_SEC_DISPLAY)))
+        onView(withId(R.id.time_stamp)).check(matches(withText(context.getString(R.string.timestamp_label, YEAR_2200 + 10500L))))
```

Comment:
These offsets would be easier to read as constants too. :)

It's hard for me to figure out the value just form glancing at it vs. TEN_SECONDS

Reply:
Updated to use `Duration` math here, to get the self-documenting `plusSeconds` and `plusMillis`.